libs/cache/file_cache.go

```diff
@@ -0,0 +1,253 @@
+package cache
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/databricks/cli/internal/build"
+	"github.com/databricks/cli/libs/log"
+)
+
+// Metrics is a local interface for tracking cache telemetry.
+type Metrics interface {
+	SetBoolValue(key string, value bool)
+}
+
+// FileCache implements the Cache interface using local disk storage.
+type FileCache[T any] struct {
+	baseDir       string
+	expiryMinutes int
+	mu            sync.Mutex
+	metrics       Metrics
+}
+
+// newFileCacheWithBaseDir creates a new file-based cache that stores data in the specified directory.
+func newFileCacheWithBaseDir[T any](baseDir string, expiryMinutes int) (*FileCache[T], error) {
+	if err := os.MkdirAll(baseDir, 0o700); err != nil {
+		return nil, fmt.Errorf("failed to create cache directory: %w", err)
+	}
+
+	fc := &FileCache[T]{
+		baseDir:       baseDir,
+		expiryMinutes: expiryMinutes,
+	}
+
+	// Clean up expired files synchronously
+	fc.cleanupExpiredFiles()
+
+	return fc, nil
+}
+
+// cleanupExpiredFiles removes expired cache files from disk based on file modification time.
+// This runs synchronously once when the cache is created.
+// Files older than expiryMinutes are deleted.
+func (fc *FileCache[T]) cleanupExpiredFiles() {
+	now := time.Now()
+	expiryDuration := time.Duration(fc.expiryMinutes) * time.Minute
+
+	err := filepath.Walk(fc.baseDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			// Log walk errors but continue cleanup
+			log.Debugf(context.Background(), "[Local Cache] cleanup: failed to access path %s: %v", path, err)
+			return nil
+		}
+
+		if info.IsDir() {
+			return nil
+		}
+
+		// Only process .json cache files
+		if filepath.Ext(info.Name()) != ".json" {
+			return nil
+		}
+
+		// Check if file is expired based on modification time
+		age := now.Sub(info.ModTime())
+		if age > expiryDuration {
+			if err := os.Remove(path); err != nil {
+				log.Debugf(context.Background(), "[Local Cache] cleanup: failed to remove expired file %s: %v", path, err)
+			} else {
+				log.Debugf(context.Background(), "[Local Cache] cleanup: removed expired file %s (age: %v)", path, age)
+			}
+		}
+
+		return nil
+	})
+	if err != nil {
+		log.Warnf(context.Background(), "[Local Cache] cleanup: failed to walk cache directory: %v", err)
```

Comment:
why Background and not passed context?

Reply:
Good catch, just an oversight, should be `ctx` indeed
https://github.com/databricks/cli/pull/3678/commits/828e41639ca81ca6ab64ae19bdc5c38ead1e67f3